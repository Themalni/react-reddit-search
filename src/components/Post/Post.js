import React, { Component } from 'react';
import { MasonryContainer } from '../Container/Container';
import { Card, CardHeader, CardTitle, CardBody, CardText, CardSmallText, CardFooter, CardImage, CardImageSmall } from '../Card/Card';
import { LinkButton } from '../Button/Button';
import { connect } from 'react-redux';
import { fetchPosts, isLoadingStateChange } from '../../actions/postsActions';
import PropTypes from 'prop-types';


class Post extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }
  componentDidMount() {
    this.forceUpdate()
  }
  truncateString(text, limit) {
    const shortened = text.indexOf(" ", limit);
    if (shortened === -1) return text;
    return text.substring(0, shortened) + " ...";
  }
  render() {
    const { posts } = this.props
    const postList = posts.length ? (
      posts.map(post => {
        return (
          <Card key={post.id}>
            <CardHeader>
              {post.subreddit_name_prefixed}
            </CardHeader>
            <CardBody>
              <CardTitle>
                {this.truncateString(post.title, 80)}
              </CardTitle>
              <CardSmallText>
                Posted by: {post.author}
              </CardSmallText>
              {post.preview ? <CardImage src={post.preview.images[0].source.url} /> : <CardImageSmall src={'https://cdn.comparitech.com/wp-content/uploads/2017/08/reddit-1.jpg'} />}
              <CardText>
                {this.truncateString(post.selftext, 200)}
              </CardText>
            </CardBody>
            <CardFooter>
              <LinkButton isBlue href={`https://www.reddit.com${post.permalink}`} target="_blank">Read more</LinkButton>
            </CardFooter>
          </Card>
        )
      })
    ) : null
    return (
      <MasonryContainer>
        {postList}
      </MasonryContainer>
    )
  }
}

Post.propTypes = {
  fetchPosts: PropTypes.func.isRequired,
  posts: PropTypes.array.isRequired
}

const mapStateToProps = (state) => {
  return {
    posts: state.posts.items,
    searchTerm: state.search.searchTerm.search,
    isLoading: state.posts.isLoading,
    postsNotFound: state.posts.postsNotFound
  }
}

export default connect(mapStateToProps, { fetchPosts, isLoadingStateChange })(Post)