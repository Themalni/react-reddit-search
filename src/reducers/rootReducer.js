import { combineReducers } from 'redux';
import { postsReducer } from './postsReducer';
import searchReducer from './searchReducer';

export default combineReducers({
  posts: postsReducer,
  search: searchReducer
})