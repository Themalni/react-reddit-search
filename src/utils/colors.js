export const colors = {
  white: '#fff',
  graphite: '#666',
  grey: '#8a96a3',
  yellowOrange: '#ff9933',
  orange: '#ed5f55',
  orangeDark: '#d8574e',
  blue: '#3a81b7',
  blueLight: '#64a0ce'
};

// d02216