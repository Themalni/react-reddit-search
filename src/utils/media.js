export const media = {
  phone: '@media (min-width: 320px)',
  tablet: '@media (min-width: 728px)',
  laptop: '@media (min-width: 1024px)',
  desktop: '@media (min-width: 1280px)'
}