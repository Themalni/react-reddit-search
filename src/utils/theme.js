import { colors } from './colors';

export const theme = {
  colors,
  font: {
    thin: 300,
    regular: 400,
    bold: 700,
  },
  media: {
    phone: '@media (min-width: 320px)',
    tablet: '@media (min-width: 728px)',
    laptop: '@media (min-width: 1024px)',
    desktop: '@media (min-width: 1280px)'
  }
}