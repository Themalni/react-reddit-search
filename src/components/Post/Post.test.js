import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import rootReducer from '../../reducers/rootReducer';
import { theme } from '../../utils/theme';
import Post from './Post';
import { ThemeProvider } from 'styled-components';

afterEach(() => cleanup())

function renderWithRedux(component,
  { initState, store = createStore(rootReducer, initState, applyMiddleware(thunk)) } = {}) {
  return {
    ...render(<Provider store={store}><ThemeProvider theme={theme}>{component}</ThemeProvider></Provider>),
    store
  }
}

describe('Search', () => {

  it("renders correctly", () => {
    const { container } = renderWithRedux(<Post />);
    expect(container.firstChild).toMatchSnapshot();
  })

})