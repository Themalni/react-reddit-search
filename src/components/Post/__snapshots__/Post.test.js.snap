// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  class="sc-htpNat bmgMfx"
/>
`;
